### ISSUE TITLE
Problem: log4j.properties missing from CentOS install in Archivematica 1.8

### ISSUE THREAD

**Issue by ross-spencer** [archivematica/Issues#475]
**Expected behaviour**

When I run a zipped bagit transfer and I look at the verify transfer and restructure for compliance job I do not see log4j configuration errors. 

**Current behaviour**

I see this in `stderr` when i look at the job:
```
log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```

**Steps to reproduce**

Initiate a transfer with a zipped bag and look at the `stderr` on `Job Verify bag, and restructure for compliance`.

**Your environment (version of Archivematica, OS version, etc)**

CentOS, Archivematica 1.8. Verified on: http://am18centos.qa.archivematica.org

**Additional context**

Fixed on http://am18centos.qa.archivematica.org by doing the following:

1. Create a `conf` folder in `/usr/share/bagit/`
2. Create a `log4j.properties` file
3. Add the contents:

> ```
> # Set root logger level to DEBUG and its appenders to CONSOLE and R.
> log4j.rootLogger=ALL, R
> 
> #File
> log4j.appender.R=org.apache.log4j.FileAppender
> log4j.appender.R.File=/tmp/bag-${log.timestamp}.log
> log4j.appender.R.layout=org.apache.log4j.PatternLayout
> log4j.appender.R.layout.ConversionPattern=%d [%t] %-5p %C{1} : %m%n
> 
> # LIMIT CATEGORIES
> log4j.logger.gov.loc.repository=INFO
> log4j.logger.org.apache.http=INFO
> ```
> 
4. Save the file
5. Permissions: Seems happy enough with `root`

> ```
> [artefactual@am18centos conf]$ pwd
> /usr/share/bagit/conf
> [artefactual@am18centos conf]$ ls -la
> total 12
> drwxr-xr-x 2 root root 4096 Feb  5 08:26 .
> drwxr-xr-x 5 root root 4096 Feb  5 08:23 ..
> -rw-r--r-- 1 root root  413 Feb  5 08:26 log4j.properties
> ```
6. Test the output in Archivematica as above. There will be no `stderr` complaining about the issue. Issues with bags should now show up as anticipated. 

Many thanks to Andy on the [forum](https://groups.google.com/d/msg/archivematica/Augr1Xo7VRs/S0meKlnNBAAJ) for reporting this and uncovering what seems to be an issue with `1.8` at least but possibly in `1.9` as well. 

---
**For Artefactual use:**
Please make sure these steps are taken before moving this issue from Review to Verified in Waffle:

- All PRs related to this issue are properly linked 👍
- All PRs related to this issue have been merged 👍
- Test plan for this issue has been implemented and passed 👍
- Documentation regarding this issue has been written and it has been added to the release notes, if needed 👍

**Comment by michaelhagedon** [comment]
Thanks, @ross-spencer! I work with Andy (@caosborne89) and have been following this issue... the `log4j.properties` you've posted here would indeed get rid of the no appenders error, which is helpful. What it doesn't fix, however, is displaying `stdout` and `stderr` from bagit in the Archivematica UI. I mention that because we actually encountered this issue when a bag failed validation and our digital preservation librarian couldn't see the reason why. So we developed the following `log4j.properties` to both display appropriate info in the UI and log to a file: 
```
# Set root logger level to DEBUG and its appenders to CONSOLE and R.
log4j.rootLogger=ALL, R, STDERR, STDOUT
 
#File
log4j.appender.R=org.apache.log4j.FileAppender
log4j.appender.R.File=/usr/share/bagit/logs/bag-${log.timestamp}.log
log4j.appender.R.layout=org.apache.log4j.PatternLayout
log4j.appender.R.layout.ConversionPattern=%d [%t] %-5p %C{1} : %m%n
 
# STDERR
log4j.appender.STDERR=org.apache.log4j.ConsoleAppender
log4j.appender.STDERR.Target=System.err
log4j.appender.STDERR.filter.e=org.apache.log4j.varia.LevelRangeFilter
log4j.appender.STDERR.filter.e.LevelMin=ERROR
log4j.appender.STDERR.filter.e.LevelMax=FATAL
log4j.appender.STDERR.layout=org.apache.log4j.PatternLayout
log4j.appender.STDERR.layout.ConversionPattern=%d [%t] %-5p %C{1} : %m%n
 
# STDOUT
log4j.appender.STDOUT=org.apache.log4j.ConsoleAppender
log4j.appender.STDOUT.Target=System.out
log4j.appender.STDOUT.filter.a=org.apache.log4j.varia.LevelRangeFilter
log4j.appender.STDOUT.filter.a.LevelMin=INFO
log4j.appender.STDOUT.filter.a.LevelMax=WARN
log4j.appender.STDOUT.layout=org.apache.log4j.PatternLayout
log4j.appender.STDOUT.layout.ConversionPattern=%d [%t] %-5p %C{1} : %m%n
 
# LIMIT CATEGORIES
log4j.logger.gov.loc.repository=INFO
log4j.logger.org.apache.http=INFO
```
I suppose our specific decisions in this config are debatable, but maybe that's enough to give Artefactual an idea of what we were wrestling with.

**Comment by michaelhagedon** [comment]
If it matters, we're using that bagit config with Archivematica 1.7.

**Comment by sromkey** [comment]
@ross-spencer want to double check if this an issue?

**Comment by jorikvankemenade** [comment]
> Many thanks to Andy on the forum for reporting this and uncovering what seems to be an issue with `1.8` at least but possibly in `1.9` as well.

I can verify that this is also a problem in version 1.9 for CentOS:

```console
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN Please initialize the log4j system properly.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN Please initialize the log4j system properly.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN Please initialize the log4j system properly.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN Please initialize the log4j system properly.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN No appenders could be found for logger (gov.loc.repository.bagit.driver.CommandLineBagDriver).
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: log4j:WARN Please initialize the log4j system properly.
Sep 02 18:22:52 archivematica-dashboard-dev.cern.ch python[7183]: VerifyChecksumsError("Unable to find expected path objects/files/92a471f1-59bf-4dc7-97ef-e6da2843d9c5-krzysztof_BSK-25.zip-2019-09-02T16_20_05.444253_00_00/krzysztof-BSK-413c507/BSK-Sprawko/._lock.Sprawozdanie.odt_ for
```

Applying the config as proposed by @ross-spencer seemed to solve it (the logging problem at least):

```console
Sep 02 19:04:55 archivematica-dashboard-dev.cern.ch python[7183]: VerifyChecksumsError("Unable to find expected path objects/files/92a471f1-59bf-4dc7-97ef-e6da2843d9c5-krzysztof_BSK-25.zip-2019-09-02T17_02_05.431159_00_00/krzysztof-BSK-413c507/BSK-Sprawko/._lock.Sprawozdanie.odt_ for
```

**Comment by ross-spencer** [comment]
@jorikvankemenade it looks like your `._lock.Sprawozdanie.odt_` file is going to be removed through this microservice [here](https://github.com/artefactual/archivematica/blob/stable/1.9.x/src/MCPClient/lib/clientScripts/remove_hidden_files_and_directories.py). I don't think we have a non-invasive workaround for you, i.e. if your file is needed, we'd recommend renaming. If the file can't be renamed, a change would need to be made to the workflow, or that microservice script. Depending on your use-case, e.g. if transferring these files as-is, is a requirement, do you want to open up a ticket describing what the problem is and what you'd like to do, and we can discuss long-term solutions? I have asked [this ticket](https://github.com/archivematica/Issues/issues/419) to be considered for `1.11` and you are more than welcome to add your notes to that one as it describes the same issue. We can rework the title a bit.

**Comment by ross-spencer** [comment]
NB. @jorikvankemenade and @michaelhagedon, the issue for #475 and log4j will not be an issue on on Archivematica `1.10` onwards as we have replaced all Java bagit usage entirely with Python bagit. 

For zipped bags on CentOS, the microservice output now looks as follows: 

![image](https://user-images.githubusercontent.com/1880412/64175773-53d08b80-ce5c-11e9-90bc-6b0996af5ffc.png)

How this impacts your use on the command line, is you'll need to run bagit from the `venv` directory associated with the MCP Client. So this will work:
```
[artefactual@am11centos /]$ /usr/share/archivematica/virtualenvs/archivematica-mcp-client/bin/python -m bagit
usage: bagit.py [-h] [--processes PROCESSES] [--log LOG] [--quiet]
                [--validate] [--fast] [--completeness-only] [--sha1]
                [--sha224] [--sha384] [--sha256] [--sha512] [--md5]
                [--source-organization SOURCE_ORGANIZATION]
                [--organization-address ORGANIZATION_ADDRESS]
                [--contact-name CONTACT_NAME] [--contact-phone CONTACT_PHONE]
                [--contact-email CONTACT_EMAIL]
                [--external-description EXTERNAL_DESCRIPTION]
                [--external-identifier EXTERNAL_IDENTIFIER]
                [--bag-size BAG_SIZE]
                [--bag-group-identifier BAG_GROUP_IDENTIFIER]
                [--bag-count BAG_COUNT]
                [--internal-sender-identifier INTERNAL_SENDER_IDENTIFIER]
                [--internal-sender-description INTERNAL_SENDER_DESCRIPTION]
                [--bagit-profile-identifier BAGIT_PROFILE_IDENTIFIER]
                directory [directory ...]
bagit.py: error: too few arguments
```

And for legacy users, it seems, running with the workaround will be one approach,  but you should be able to run Python bagit in a similar way to the one I've documented here as well. 

To see where your Archivematica MCP Client `venv` is running from you can ask the MCP Client service about itself:
```
$ service archivematica-mcp-client status
```
```
Redirecting to /bin/systemctl status archivematica-mcp-client.service
● archivematica-mcp-client.service - Archivematica MCP Client Service
   Loaded: loaded (/etc/systemd/system/archivematica-mcp-client.service; enabled; vendor preset: disabled)
   Active: active (running) since Thu 2019-08-22 20:40:57 CEST; 1 weeks 4 days ago
 Main PID: 25898 (python)
   CGroup: /system.slice/archivematica-mcp-client.service
           └─25898 /usr/share/archivematica/virtualenvs/archivematica-mcp-client/bin/python   <--running here...
```
The last line will be of relevance to readers of this ticket. 

Closing this ticket as resolved in `1.10`. The [forum](https://groups.google.com/forum/#!forum/archivematica) might be a good place for any follow-up unless of course there are any unanticipated bugs folks are running into with the new mechanism.
